### PR TITLE
Locale enhancements

### PR DESCRIPTION
**Pull request by Cyanoxygen**
Description
------

This PR adds two enhancements:
- zh_CN localization
- Set locale settings of the container according to system locale

I do not have full build tools installed yet, so changes in the code are not tested.

Known issues
------

~~App is showing question marks instead of Chinese characters when launched with zh_CN locale.~~ This is worked around by commit f5537f6.

### CODE REVIEW

**Comment by KexyBiscuit** [comment]
Known issue is also in the upstream: microsoft/WSL-DistroLauncher#44

**Comment by KexyBiscuit** [comment]
Known issue resolved, have no idea what's the side affect so not opening PR to microsoft side now.

File: AOSCOSLauncher/AOSCOSLauncher.cpp [hunk 1]
```diff
@@ -84,8 +101,43 @@ HRESULT SetDefaultUser(std::wstring_view userName)
...
+    LPWSTR locale = (LPWSTR)malloc(LOCALE_NAME_MAX_LENGTH * 2);
+    if (locale == nullptr) {
+        // Failed to get a buffer, we have nothing to do but bailing out.
+        return CO_E_INIT_MEMORY_ALLOCATOR;
+    }
+    result = GetUserDefaultLocaleName(locale, LOCALE_NAME_MAX_LENGTH);
+    if (result <= 0) {
+        // GetUserDefaultLocaleName() returns the string length of the locale name.
+        // If it fails, fallback to en_US.
+        hr = StringCchCopyW(locale, LOCALE_NAME_MAX_LENGTH, L"en_US");
```
Comment (KexyBiscuit) [inline AOSCOSLauncher/AOSCOSLauncher.cpp:123]:
```suggestion
        hr = StringCchCopyW(locale, LOCALE_NAME_MAX_LENGTH, L"en-US");

```

Windows Locale Names are based on [RFC 4646](https://datatracker.ietf.org/doc/rfc4646/), so subtags are distinguished and separated from one another by a hyphen.

File: AOSCOSLauncher/messages.mc [hunk 2]
```diff
@@ -1,25 +1,47 @@
...
+LanguageNames = (zh_CN=0x804:MSG00804)
+
 
 MessageId=1001 SymbolicName=MSG_WSL_REGISTER_DISTRIBUTION_FAILED
 Language=English
 WslRegisterDistribution failed with error: 0x%1!x!
 .
 
+Language=zh_CN
+方法 WslRegisterDistribution 执行时出错: 0x%1!x!
```
Comment (KexyBiscuit) [inline AOSCOSLauncher/messages.mc:11]:
```suggestion
函数 WslRegisterDistribution 执行时出错: 0x%1!x!
```

File: AOSCOSLauncher/messages.mc [hunk 3]
```diff
@@ -1,25 +1,47 @@
...
+.
+
+
 MessageId=1002 SymbolicName=MSG_WSL_CONFIGURE_DISTRIBUTION_FAILED
 Language=English
 WslGetDistributionConfiguration failed with error: 0x%1!x!
 .
 
+Language=zh_CN
+方法 WslGetDistributionConfiguration 执行时出错: 0x%1!x!
```
Comment (KexyBiscuit) [inline AOSCOSLauncher/messages.mc:21]:
```suggestion
函数 WslGetDistributionConfiguration 执行时出错: 0x%1!x!
```

File: AOSCOSLauncher/messages.mc [hunk 4]
```diff
@@ -1,25 +1,47 @@
...
+.
+
+
 MessageId=1003 SymbolicName=MSG_WSL_LAUNCH_INTERACTIVE_FAILED
 Language=English
 WslLaunchInteractive %1 failed with error: 0x%2!x!
 .
 
+Language=zh_CN
+方法 WslLaunchInteractive 执行命令 `%1` 时出错: 0x%2!x!
```
Comment (KexyBiscuit) [inline AOSCOSLauncher/messages.mc:31]:
```suggestion
函数 WslLaunchInteractive 执行命令 `%1` 时出错: 0x%2!x!
```

File: AOSCOSLauncher/messages.mc [hunk 5]
```diff
@@ -1,25 +1,47 @@
...
+.
+
+
 MessageId=1004 SymbolicName=MSG_WSL_LAUNCH_FAILED
 Language=English
 WslLaunch %1 failed with error: 0x%2!x!
 .
 
+Language=zh_CN
+方法 WslLaunch 执行命令 `%1` 时出错: 0x%2!x!
```
Comment (KexyBiscuit) [inline AOSCOSLauncher/messages.mc:41]:
```suggestion
函数 WslLaunch 执行命令 `%1` 时出错: 0x%2!x!
```

File: AOSCOSLauncher/messages.mc [hunk 6]
```diff
@@ -47,45 +69,123 @@ Usage:
...
+
+
 MessageId=1013 SymbolicName=MSG_INSTALL_ALREADY_EXISTS
 Language=English
 The distribution installation has become corrupted.
 Please select Reset from App Settings or uninstall and reinstall the app.
 .
+
+Language=zh_CN
+此发行版的实例将被损坏。
```
Comment (KexyBiscuit) [inline AOSCOSLauncher/messages.mc:179]:
```suggestion
此发行版的实例已被损坏。
```

**Review by KexyBiscuit — CHANGES_REQUESTED** [review]
Cannot really read zh-CN, just a few changes.

File: AOSCOSLauncher/AOSCOSLauncher.cpp (hunk 1 above)
Comment (Cyanoxygen) [reply AOSCOSLauncher/AOSCOSLauncher.cpp:123]: `locale` is later used by Linux, so it needs to be underscored before formatting to actual command. I will fix this later.

**Comment by Cyanoxygen** [comment]
Just realized that these locale codes are not compatible with each other sometimes. We need a implementation which ignores invalid codes to fallback to en_US, which is what I'm thinking about.

**Comment by Cyanoxygen** [comment]
> Or, we can ignore this, just replace any hyphens with an underscore and format this into the command line, which will be ignored by Linux if the language code is invalid.

Yes, after some research, we can just take the replace approach instead.

**Review by KexyBiscuit — APPROVED** [review]

**Pull request merged**